src/common/values.ts

```diff
@@ -102,29 +103,29 @@ const writeValuesToFile = async (
   const d = terminal('common:values:writeValuesToFile')
   const isSecretsFile = targetPath.includes('/secrets.') && hasSops
   const values = cloneDeep(inValues)
-  const newValues = removeBlankAttributes(values)
-  if (isEmpty(newValues) && isSecretsFile && overwrite) {
+  const cleanedValues = removeBlankAttributes(values)
+  if (isEmpty(cleanedValues) && isSecretsFile) {
     // get rid of empty secrets files as those are problematic
-    if (existsSync(targetPath)) unlinkSync(targetPath)
-    if (existsSync(`${targetPath}.dec`)) unlinkSync(`${targetPath}.dec`)
+    if (await pathExists(targetPath)) await unlink(targetPath)
+    if (await pathExists(`${targetPath}.dec`)) await unlink(`${targetPath}.dec`)
+    return
   }
-  d.debug('newValues: ', JSON.stringify(newValues, null, 2))
   const suffix = isSecretsFile ? '.dec' : ''
-  if (!existsSync(targetPath) || overwrite) {
+  if (!(await pathExists(targetPath)) || overwrite) {
     // create the non-suffixed file for encryption to not skip this later on
-    const notExists = !existsSync(targetPath)
+    const notExists = !(await pathExists(targetPath))
     if (isSecretsFile && notExists) {
-      await writeFile(targetPath, objectToYaml(newValues))
+      await writeFile(targetPath, objectToYaml(values))
       await encrypt(targetPath)
       await decrypt(targetPath)
       return
     }
-    await writeFile(targetPath + suffix, objectToYaml(newValues))
+    await writeFile(targetPath + suffix, objectToYaml(values))
```

Comment:
before values with removed blank attributes were saved but now you decided to save values with blank attributes. Why?

Reply:
to stay consistent between core and api to avoid diff